### PR TITLE
fix: add carousel component of demo

### PR DESCRIPTION
**Pull request by James-9696**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Introduced a new carousel card display using the Composition API.
  - Added a dialog box component with carousel integration and interactive buttons.

- **Enhancements**
  - Updated carousel demo configuration to include new features "Carousel Card" and "Dialog Show" with multilingual descriptions.

- **Testing**
  - Added tests for carousel card functionality and dialog showcase using Playwright.

### CODE REVIEW

File: examples/sites/demos/pc/app/carousel/card-show-composition-api.vue [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  </tiny-carousel>
+</template>
+
+<script setup>
+import { ref, reactive } from 'vue'
+import { Carousel as TinyCarousel, CarouselItem as TinyCarouselItem, Card as TinyCard } from '@opentiny/vue'
+
+let curIndex = ref(0)
+const dsj = ref(`/static/images/dsj.png`)
+const userHead = ref(`/static/images/user-head.png`)
```
Comment (zzcr) [inline examples/sites/demos/pc/app/carousel/card-show-composition-api.vue]: 新增的示例，也需要补充E2E测试用例

File: examples/sites/demos/pc/app/carousel/webdoc/carousel.js [hunk 2]
```diff
@@ -171,6 +171,30 @@ export default {
...
       codeFiles: ['carousel-events.vue']
+    },
+    {
+      demoId: 'card-show',
+      name: {
+        'zh-CN': '轮播卡片',
+        'en-US': 'Carousel Card'
+      },
+      desc: {
+        'zh-CN': '<p>通过设置<code>default</code>插槽，自定义卡片轮播场景。</p>',
```
Comment (gimmyhehe) [inline examples/sites/demos/pc/app/carousel/webdoc/carousel.js]: 文档整改规范里面，这个code两侧要加一个空格~ 
如 设置 \<code\>default\<\/code\> 插槽

File: examples/sites/demos/pc/app/carousel/card-show-composition-api.vue [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    </tiny-carousel-item>
+  </tiny-carousel>
+</template>
+
+<script setup>
+import { ref, reactive } from 'vue'
+import { Carousel as TinyCarousel, CarouselItem as TinyCarouselItem, Card as TinyCard } from '@opentiny/vue'
+
+let curIndex = ref(0)
+const dsj = ref(`/static/images/dsj.png`)
```
Comment (gimmyhehe) [inline examples/sites/demos/pc/app/carousel/card-show-composition-api.vue]: 静态资源路径不能写死，需要改成`${import.meta.env.VITE_APP_BUILD_BASE_URL}static/images/dsj.png`

Comment (James-9696) [reply examples/sites/demos/pc/app/carousel/card-show-composition-api.vue]: done

File: examples/sites/demos/pc/app/carousel/webdoc/carousel.js (hunk 2 above)
Comment (James-9696) [reply examples/sites/demos/pc/app/carousel/webdoc/carousel.js]: done

**Review by zzcr — APPROVED** [review]

**Review by zzcr — APPROVED** [review]

**Pull request merged**